src/Api/Commands/SplitToDiscussionHandler.php

```diff
@@ -0,0 +1,197 @@
+
+        $command->actor->assertCan('split', $startPost->discussion);
+
+        /** @var Discussion $originalDiscussion */
+        $originalDiscussion = $startPost->discussion;
+
+        $discussion = Discussion::find($command->discussion_id);
+
+        // update all posts that are split.
+        $affectedPosts = $this->assignPostsToDiscussion(
+            $originalDiscussion,
+            $discussion,
+            $startPost->number,
+            $command->end_post_number
+        );
+
+        $this->renumberDiscussion($originalDiscussion);
+        $originalDiscussion = $this->refreshDiscussion($originalDiscussion);
+
+        $this->renumberDiscussion($discussion);
+        $discussion = $this->refreshDiscussion($discussion);
+
+        $this->events->dispatch(
+            new DiscussionWasSplit($command->actor, $affectedPosts, $originalDiscussion, $discussion)
+        );
+
+        return $discussion;
+    }
+
+    /**
+     * Assign the specific range to a new Discussion without resetting post numbers.
+     *
+     * @param Discussion $originalDiscussion
+     * @param Discussion $discussion
+     * @param            $startPostNumber
+     * @param            $endPostNumber
+     *
+     * @return Collection
+     */
+    protected function assignPostsToDiscussion(
+        Discussion $originalDiscussion,
+        Discussion $discussion,
+        $startPostNumber,
+        $endPostNumber
+    ): Collection {
+        $lastPostNumber = $this->posts
+            ->query()
+            ->where('discussion_id', $discussion->id)
+            ->max('number');
+
+        $this->posts
+            ->query()
+            ->where('discussion_id', $originalDiscussion->id)
+            ->whereBetween('number', [$startPostNumber, $endPostNumber])
+            ->each(function (Post $post) use (&$lastPostNumber, $discussion) {
+                $lastPostNumber++;
+                $post->discussion_id = $discussion->id;
+                $post->number = $lastPostNumber;
+                $post->save();
+            });
+
+        // Update relationship posts on new discussion.
+        $discussion->load('posts');
+
+        $discussion->post_number_index = $lastPostNumber;
+        $discussion->save();
+
+        return $discussion->posts;
+    }
+
+    /**
+     * Re-assign numbers starting from one to a discussion.
+     *
+     * @param Discussion $discussion
+     */
+    protected function renumberDiscussion(Discussion $discussion)
+    {
+        $sql = '
+            SET @maxNumber = (SELECT MAX(number) FROM posts WHERE discussion_id = '.$discussion->id.');
+            SET @rank = @maxNumber;
+            UPDATE posts SET number=@rank:=@rank+1 WHERE discussion_id = '.$discussion->id.' ORDER BY created_at;
+            UPDATE posts SET number=number-@maxNumber WHERE discussion_id = '.$discussion->id.';
+        ';
+
+        resolve('db')->unprepared($sql);
```

Comment:
This piece of code screams for SQL injection... 😕

Reply:
Yes, this needs to be refactored into Eloquent code. Best would probably be to use a prepared statement as well.

Reply:
As others have said, this is just a huge red flag. All of this can be handled using the Eloquent code.